.github/workflows/build-and-test.yml

```diff
@@ -943,6 +942,104 @@ jobs:
         if: ${{ failure() && runner.debug }}
         uses: canonical/action-tmate@main
 
+  juju-adopt-ceph-test:
+    needs:
+      - build
+    name: Adopt Charmed ceph cluster test
+    runs-on: self-hosted-linux-amd64-noble-xlarge
+    steps:
+      - name: Download charm
+        uses: actions/download-artifact@v4
+        with:
+          name: charms
+          path: ~/artifacts/
+
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Setup LXD
+        uses: canonical/setup-lxd@v0.1.1
+        with:
+          # pin lxd to LTS release.
+          channel: 5.21/stable
+
+      - name: Install dependencies
+        run: ./tests/scripts/ci_helpers.sh install_deps
+
+      - name: Install MicroCeph charm
+        run: |
+          set -eux
+          date
+          juju deploy ./tests/bundles/noble-caracal.yaml
+          juju wait-for application ceph-mon --query='forEach(units, unit => unit.workload-status=="active" && unit.agent-status=="idle")' --timeout=30m
+          juju wait-for application secondary --query='forEach(units, unit => unit.workload-status=="active" && unit.agent-status=="idle")' --timeout=30m
+          date
+
+      - name: Adopt the charmed ceph cluster using primary microcpeh application
+        run: |
+          set -eux
+          date
+          juju integrate ceph-mon primary
```

Comment:
Nit: to make it super clear which interface we're adding maybe specify the full rel name